packages/ui/src/components/editor/panels/Assets/container/index.tsx

```diff
@@ -473,8 +486,8 @@ const AssetPanel = () => {
     // TODO: add settings functionality
   }
 
-  const handleUpdateAsset = () => {
-    // TODO: add upload asset functionality
+  const refreshDirectory = async () => {
```

Comment:
My apologies for the confusion, my previous wasn't a change request but more like a question... Based on the fact that I'm still unfamiliar with the system I wanted to bring up that maybe assetsQuery.refetch() was missing an await just because the wrapper function was defined as an async function.

But I just went through the code in order to confirm if refetch needed to be awaited or not, I found that refetch is a reference of a function named fetch() which isn't defined as a promise, therefore neither assetsQuery.refetch() requires to be awaited, nor refreshDirectory requires to be defined as a promise.

```ts
  const refreshDirectory = () {
    assetsQuery.refetch()
  }
```
I still have a question though, why do we need to wrap assetsQuery.refetch() within a new function when nothing else is being done within that function (refreshDirectory), can't we simply directly use assetsQuery.refetch() instead?

Example

```ts
onClick={() =>
            inputFileWithAddToScene({
              projectName: originalPath as string,
              directoryPath: `projects/${originalPath}/assets/`
            })
              .then(assetsQuery.refetch)
              .catch((err) => {
                NotificationService.dispatchNotify(err.message, { variant: 'error' })
              })
          }
```